packages/hurl/src/http/curl_cmd.rs

```diff
@@ -90,11 +95,13 @@ fn method_params(request_spec: &RequestSpec) -> Vec<String> {
     request_spec.method.curl_args(has_body)
 }
 
-/// Returns the curl args corresponding to the HTTP headers, from a request spec.
-fn headers_params(request_spec: &RequestSpec) -> Vec<String> {
+/// Returns the curl args corresponding to the HTTP headers, from a request spec and a list of raw headers from client options.
+fn headers_params(request_spec: &RequestSpec, options_headers: &[&str]) -> Vec<String> {
```

Comment:
The code here is like what we have to do in `client.rs`:  we want to change the signature of the method `header_params` from:

```rust
headers_params(request_spec: &RequestSpec) -> Vec<String>
```

to 

```rust
headers_params( headers: &HeaderVec, implicit_content_type: Option<&str>,) -> Vec<String>
```

We will do the aggregation _prior_ calling `headers_params`. In the point of view of `headers_params`, it only receive a list of headers, it doen't have to know if the headers comes from the spec (the file) or the command line.

So we do the aggregation before calling `headers_params` (exactly like we do in `client.rs`, there is a bit of redundancy but it's acceptable for the moment.

```rust
impl CurlCmd {
    /// Creates a new curl command, based on an HTTP request, cookies, a context directory, output
    /// and runner options.
    pub fn new(
        request_spec: &RequestSpec,
        cookies: &[Cookie],
        context_dir: &ContextDir,
        output: Option<&Output>,
        options: &ClientOptions,
    ) -> Self {
        let mut args = vec!["curl".to_string()];
       // ...
        let options_headers = options
            .headers
            .iter()
            .map(|h| h.as_str())
            .collect::<Vec<&str>>();
        let headers = &request_spec.headers.aggregate_raw_headers(&options_headers);
        let mut params = headers_params(headers, rrequest_spec.implicit_content_type.as_deref())
```

Note: we'er changing also the name of the variable / parameter: it's not `let headers_spec`, it's now `let headers` we "notify"/indicate that the header is a general aggragation and is not comming only form the "spec" (= the file)

Reply:
@jcamiel We might still need to keep `request_spec` for `headers_params` since we're also setting the content type based on the request body here:

https://github.com/Orange-OpenSource/hurl/blob/1a13135638ff959bb87f1aebd1236482b6c36a40/packages/hurl/src/http/curl_cmd.rs#L113-L128

Reply:
Maybe have a `has_body` bool param in the method (alongside the optional implicit content type) could be sufficient?

Reply:
But we still need to match the type of `request_spec.body`, right?

Reply:
Ah yes sorry, read the code too quickly. So I would pass the body has a reference for a start. I prefer to do this (instead of the request_spec) because it limits what you pass to the method (a body instead of the whole request)